packages/react-components/src/components/utils/SendBoxUtils.ts

```diff
@@ -162,6 +162,31 @@ export const cancelInlineImageUpload = (props: CancelInlineImageUploadProps): vo
   }
 };
 
+/* @conditional-compile-remove(rich-text-editor-image-upload) */
+interface InsertInlineImageProps {
+  imageSrcArray: string[] | undefined;
+  inlineImages: AttachmentMetadataInProgress[] | undefined;
+  messageId?: string;
+  editBoxOnInsertInlineImage?: (imageUrl: string, imageFileName: string, messageId: string) => void;
+  sendBoxOnInsertInlineImage?: (imageUrl: string, imageFileName: string) => void;
+}
+
+/* @conditional-compile-remove(rich-text-editor-image-upload) */
+/**
+ * @internal
+ */
+export const insertInlineImage = (props: InsertInlineImageProps): void => {
+  const { imageSrcArray, inlineImages, messageId, editBoxOnInsertInlineImage, sendBoxOnInsertInlineImage } = props;
```

Comment:
In our API change conversation yesterday, we decided to change this imageSrcArray( an array of what's currently in the segments after changes) to deletedImages (or changedImages to suit this case).
The issue will be that, this function is also called when delete related changes occurs.
If user delete image A, and in the previous onCancel that's called before this onInsert, the deleted image is already removed from inlineImages. Now !inlineImages.find((uploadImage) => uploadImage.url === image will be true and we will add back the deleted image.

Reply:
But in this case imageSrcArray doesn't have this image anymore (as it's not in content anymore) and `!inlineImages.find((uploadImage) => uploadImage.url === image` will never be true because it's not iterated over the deleted element